components/markdown_image.jsx

```diff
@@ -6,43 +6,100 @@ import React from 'react';
 
 import ExternalImage from 'components/external_image';
 import SizeAwareImage from 'components/size_aware_image';
+import ViewImageModal from 'components/view_image';
+
+export default class MarkdownImage extends React.PureComponent {
+    static defaultProps = {
+        imageMetadata: {},
+    };
+
+    static propTypes = {
+        alt: PropTypes.string,
+        imageMetadata: PropTypes.object,
+        src: PropTypes.string.isRequired,
+        title: PropTypes.string,
+        className: PropTypes.string.isRequired,
+        postId: PropTypes.string.isRequired,
+        imageIsLink: PropTypes.bool.isRequired,
+        onImageLoaded: PropTypes.func,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showModal: false,
+        };
+    }
+
+    showModal = (e) => {
+        if (!this.props.imageIsLink) {
+            e.preventDefault();
+            this.setState({showModal: true});
+        }
+    }
+
+    hideModal = () => {
+        this.setState({showModal: false});
+    }
+
+    render() {
+        const {imageMetadata, src, imageIsLink} = this.props;
+
+        return (
+            <ExternalImage
+                src={src}
+                imageMetadata={imageMetadata}
+            >
+                {(safeSrc) => {
+                    if (!safeSrc) {
+                        return (
+                            <a
+                                className='theme markdown__link'
+                                href={src}
+                                rel='noopener noreferrer'
+                                target='_blank'
+                                title={this.props.title}
+                            >
+                                {this.props.alt}
+                            </a>
+                        );
+                    }
+
+                    const className = imageIsLink ?
+                        this.props.className :
+                        `${this.props.className} markdown-inline-img--hover cursor--pointer a11y--active`;
+
+                    const getFileExtentionFromUrl = (url) => url.substring(safeSrc.lastIndexOf('.') + 1);
```

Comment:
On my local machine when I set the `Enable Image Proxy` setting to `true`, I was getting empty `imageMetadata`, but an extension is required for the preview window to work. I'm not sure if this is only an issue on my local environment. Therefore, I opted fro this simple solution. Don't know if it is worth adding a more robust utility function instead.